src/ApiKey/ApiKeyAuthenticationException.php

```diff
@@ -4,4 +4,23 @@
 
 class ApiKeyAuthenticationException extends \Exception
 {
+    public static function cultureFeedErrorForKey(ApiKey $apiKey): ApiKeyAuthenticationException
```

Comment:
I like the idea of having factory methods for the messages, but in this case we would be coupling `CultureFeed` (the underlying backend) to the abstraction of api keys (and their exceptions).
Can you find a way to do it without making it specific to culturefeed here?

Reply:
Done, take a look now

Reply:
Yep looks great, I like this solution a lot! 👍